fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java

```diff
@@ -581,6 +582,21 @@ public void validateForCreate(final String json, final boolean isMeetingMandator
         }
 
         validatePartialPeriodSupport(interestCalculationPeriodType, baseDataValidator, element, loanProduct);
+
+        // validate enable installment level delinquency
+        if (this.fromApiJsonHelper.parameterExists(LoanProductConstants.ENABLE_INSTALLMENT_LEVEL_DELINQUENCY, element)) {
+            if (loanProduct.getDelinquencyBucket() == null) {
+                baseDataValidator.reset().parameter(LoanProductConstants.ENABLE_INSTALLMENT_LEVEL_DELINQUENCY).failWithCode(
+                        "supported.for.loan.products.with.delinquency.bucket",
+                        "Installment level delinquency is supported for loan products with delinquency bucket");
```

Comment:
I am not sure this error message explains the problem if installment level delinquency is enabled but no delinquency bucket is set. Can you please double check whether we can explain this more in details to avoid confusions?